### ISSUE TITLE
No file system events observed when not using recursive mode

### ISSUE THREAD

**Issue by cedrickring** [gorakhargosh/watchdog#918]
We tried to listen for events on a single file, without the need for watching recursively. However without specifying `recursive=True`, no events were observed.

Platform: macOS 12.6 & Ubuntu 18.04
Python: 3.9.14

Code to reproduce:
```
import sys
import time
import logging
from watchdog.observers import Observer
from watchdog.events import LoggingEventHandler

if __name__ == "__main__":
    logging.basicConfig(level=logging.INFO,
                        format='%(asctime)s - %(message)s',
                        datefmt='%Y-%m-%d %H:%M:%S')
    path = sys.argv[1] if len(sys.argv) > 1 else '.'
    event_handler = LoggingEventHandler()
    observer = Observer()
    observer.schedule(event_handler, path)
    observer.start()
    try:
        while True:
            time.sleep(1)
    finally:
        observer.stop()
        observer.join()
```
and launching with: `python file.py <path to single file>`.

Also using the `FSEventsObserver` on macOS didn't change the outcome. Only specifying `recursive=True` did.

**Comment by cedrickring** [comment]
I totally agree with your points. However, the docs should point out that you always need to watch recursively or just default to it on darwin machines...

**Comment by Rosuav** [comment]
Agreed. I got bitten by this today; a script that worked absolutely fine on my Linux box (Debian Bullseye, inotify) utterly failed to report anything on my client's Mac. Setting `recursive=True` made it work. Having this be the default when using FSEvents would solve that issue.

**Comment by Rosuav** [comment]
I'll have a whack at it, but I don't have a Mac to test on, so all I can say is, the tests haven't started failing on Linux.

https://github.com/gorakhargosh/watchdog/pull/1013

**Comment by JonathanWylie** [comment]
I think  recursive=False should work, see: https://github.com/gorakhargosh/watchdog/pull/779
What I found in my case was that the paths that fsevents reports are the real paths, if I am watching via a symlink, then the paths don't match and _is_recursive_event ends up excluding everything.
